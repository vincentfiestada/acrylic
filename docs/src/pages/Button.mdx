import {Button} from "../../../lib/acrylic";

## Button
<Button>Button</Button>
