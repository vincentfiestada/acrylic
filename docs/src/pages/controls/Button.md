## Button

A **Button** is a basic control that allows the user to interact via clicks and taps.

```jsx
render(() => (
	<div>
		<Button>Button</Button>
		<Button variant="primary">Primary</Button>
		<Button variant="success">Success</Button>
		<Button variant="warning">Warning</Button>
		<Button variant="danger">Danger</Button>
		<Button disabled>Disabled</Button>
	</div>
))
```

Buttons can also contain **[Icons](/components/typography/Icon)**.

```jsx
render(() => (
	<div>
		<Button><Icon src="/img/ok.svg" />Okay</Button>
		<Button><Icon src="/img/cancel.svg" />Cancel</Button>
		<Button><Icon src="/img/about.svg" />About</Button>
	</div>
))
```

Use an **IconButton** if you don't need to display text.

```jsx
render(() => (
	<div>
		<IconButton src="/img/ok.svg" />
		<IconButton src="/img/cancel.svg" />
		<IconButton src="/img/about.svg" />
	</div>
))
```

###### Documentation
- [`Button`](/wiki/modules/_components_controls_button_.html)
- [`IconButton`](/wiki/modules/_components_controls_iconbutton_.html)
- [`Variant`](/wiki/modules/_components_variant_.html)
