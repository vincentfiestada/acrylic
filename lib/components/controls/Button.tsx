import * as React from "react";
import IProps from "../IProps";

export interface IButtonProps extends IProps {
	children: React.ReactNode; // Button must have children
}

export default function Button(props: IButtonProps) {
	const { children, ...otherProps } = props;
	return <button {...otherProps}>{children}</button>;
}
