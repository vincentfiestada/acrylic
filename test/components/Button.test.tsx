import * as React from "react";
import Button from "../../lib/components/controls/Button";

describe("<Button />", () => {
	it("Should match the latest snapshot", () => {
		expect(<Button>A Button</Button>).toMatchSnapshot();
	});
});
