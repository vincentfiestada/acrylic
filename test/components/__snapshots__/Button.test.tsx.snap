// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should match the latest snapshot 1`] = `
<Button>
  A Button
</Button>
`;
